src/lib/core/datetime.c

```diff
@@ -364,6 +362,10 @@ datetime_increment_by(struct datetime *self, int direction,
 	bool is_ym_updated = false;
 	int years = ival->year;
 	int months = ival->month;
+	int weeks = ival->week;
+	int days = ival->day;
+	int hours = ival->hour;
+	int minutes = ival->min;
 	int seconds = ival->sec;
```

Comment:
You use `int` for seconds here and `double` below. Which is right?

Reply:
I've fixed this problem - using `int64_t` in the patch above and `double` in the structure. Same for `min`, `hour` and `date`. And `years`, `months`, and `weeks` - because it makes range checking code simpler. (We need to check arithmetic on 32-bit boundary violations, and we have to proceed those checks in 64-bit mode)